### PR TITLE
QUA-519: add regex filter docs

### PR DESCRIPTION
**Pull request by dantevvp**
This PR adds some documentation on the `$regex` filter parameter.

## Filterable resources overview

image

## Get files endpoint

image

## Related issue

https://codeclimate.atlassian.net/browse/QUA-519

### CODE REVIEW

File: source/includes/_overview.html.md [hunk 1]
```diff
@@ -111,7 +122,7 @@ The JSON API is agnostic beyond the `filter[]` key. In the Code Climate API,
 filters can be of two basic formats (examples on the right):
 
 1. A basic key/value
-2. A more complex data structure with an operator. At this time only the `$in` operator is supported.
+2. A more complex data structure with an operator.
```
Comment (f-moya) [inline source/includes/_overview.html.md]:
```suggestion
2. A more complex data structure with an operator. At this time only the `$in` and `$regex` operators are supported.
```

maybe ?

File: source/includes/_repository_analysis.html.md [hunk 2]
```diff
@@ -258,7 +258,13 @@ Sorted by path in ascending order.
...
 ### Query Parameters
 
-[Paginated](#collection-pagination)
+[Paginated](#collection-pagination), [Filterable](#collection-filtering)
+
+Filters include:
+
+| Name | Description | Required? |
+| ---- | ----------- | --------- |
+| filter[path] | Single path, `$in` clause containing list of categories or `$regex` clause containing regular expressions | No |
```
Comment (f-moya) [inline source/includes/_repository_analysis.html.md:267]: are you sure the `$in` operator works for this field too ?

**Review by f-moya — APPROVED** [review]
Two minor comments but it looks good 👍🏼

**Comment by f-moya** [comment]
@dantevvp Can we wait for this one https://github.com/codeclimate/slate/pull/18 to be merged ? CI will fail on master otherwise

File: source/includes/_repository_analysis.html.md (hunk 2 above)
Comment (dantevvp) [reply source/includes/_repository_analysis.html.md:267]: Yes, it works as well 👍

**Pull request merged**